src/Resolver/File.php

```diff
@@ -81,6 +81,16 @@ public function resolve($definition)
             $path = realpath($this->getIterator()->getRootDefinition()->getBasepath() . \DIRECTORY_SEPARATOR . $path);
         }
 
-        return file_get_contents($path);
+        $content = file_get_contents($path);
+
+        if ($parse == 'auto' && pathinfo($path, PATHINFO_EXTENSION) == 'json') {
```

Comment:
Do we need type coercion here? Also missed explicit json parsing (edge case where json might be in a `json.txt` file maybe?).
```suggestion
        if (($parse === 'auto' && pathinfo($path, PATHINFO_EXTENSION) === 'json') || $parse === 'json') {
```

Reply:
I think that's actually an ambiguity in the upward spec: https://github.com/magento-research/pwa-studio/tree/release/2.0/packages/upward-spec#fileresolver-configuration-options. I interpreted that to mean that the acceptable values for `parse` are either `"text"` or `"auto"`, but the paragraph after implies that other values (including `"json"`) *may* be acceptable.

Reply:
Confirmed with the architect, explicit json parsing is an expected feature. I'll review the `upward-spec` docs and suggest an edit if there's a change that would make that more clear.